bcipy/task/orchestrator/orchestrator.py

```diff
@@ -69,28 +77,71 @@ def __init__(
         self.visualize = visualize
         self.progress = 0
 
-        self.ready_to_execute = True
+        self.ready_to_execute = False
         self.logger.info("Session Orchestrator initialized successfully")
 
     def add_task(self, task: Type[Task]) -> None:
+        """Add a task to the orchestrator"""
         self.tasks.append(task)
         self.task_names.append(task.name)
+        self.ready_to_execute = True
 
     def add_tasks(self, tasks: List[Type[Task]]) -> None:
+        """Add a list of tasks to the orchestrator"""
         for task in tasks:
             self.add_task(task)
+        self.ready_to_execute = True
```

Comment:
I don't think this `self.ready_to_execute` is needed since this method is doing a self send to a method that already sets this.

Reply:
This is to prevent execution without any Tasks defined; before execution has occurred, more tasks can be added via a list of Tasks or a single Task, and then all tasks will be reset. This allows a caller to continue using the orchestrator after completing some protocol. I will rethink this as it would technically constitute a new/modified protocol.